wagtailmarkdown/static/wagtailmarkdown/js/markdown-textarea-adapter.js

```diff
@@ -0,0 +1,44 @@
+(function() {
+    function MarkdownTextarea(html, config) {
+        this.html = html;
+        this.baseConfig = config;
+    }
+    MarkdownTextarea.prototype.render = function(placeholder, name, id, initialState) {
+        placeholder.outerHTML = this.html.replace(/__NAME__/g, name).replace(/__ID__/g, id);
+
+        var element = document.getElementById(id);
+        element.value = initialState;
+
+        easymdeAttach(id);
+
+        // define public API functions for the widget:
+        // https://docs.wagtail.io/en/latest/reference/streamfield/widget_api.html
+        return {
```

Comment:
According to https://docs.wagtail.io/en/stable/reference/streamfield/widget_api.html, the object returned should have an `idForLabel` attribute.

Reply:
heh, good spot. added.

it works without it. I read that as not providing it or setting it as null to be the same thing

Reply:
Compared to https://github.com/wagtail/wagtail/blob/cdeed1954e3498ea2acab1ca43b2bdffb23af56e/client/src/entrypoints/admin/telepath/widgets.js#L9, it looks like it sets its own `idForLabel` based on the `id` provided. Clearly null `idForLabel` is valid, but based on https://github.com/wagtail/wagtail/blob/cee245e7015b8462f2aa921e42803cd7d0962b8b/client/src/components/StreamField/blocks/StructBlock.js#L70, I suspect it might be worse for accessibility if the label is missing that `for`. Think it might be worth setting this to the id.

Reply:
The markdown block does not provide a label as the main input is a hidden text area and the visual bit is a rendered editor. The block, in many ways is similart to the rich text block ones. The Draftail widget does not provide an `idForLabel` either - https://github.com/wagtail/wagtail/blob/cdeed1954e3498ea2acab1ca43b2bdffb23af56e/client/src/entrypoints/admin/telepath/widgets.js#L150-L183.

one for @thibaudcolas to hopefully guide us on. for the time being will stick to the null value and update it later to follow `DraftailRichTextArea`

Reply:
Ah I see - my mistake, sorry about that!